packages/donations-app/app/components/checkout/checkoutController.js

```diff
@@ -89,269 +54,68 @@ angular.module('aliceApp')
+      vm.donation.amount = (vm.amountOther ? vm.amountOtherValue * 100 : vm.donation.amount) || 0;
+      if (vm.donation.amount === 0) {
+        vm.noAmount = true;
+        return;
+      }
 
-    var postDonationConfirmation = function () {
-      if (AuthService.getLoggedUser()) {
-        $state.go('my-impact');
-      } else {
-        // User made an anonymous donation
-        // So he currently has no privelleges to see his impact
-        $state.go('project', {projectCode: $stateParams.projectCode});
+      if ((vm.mode == 'CARD' && !vm.cardForm.$valid) || (vm.mode == 'BANK_TRANSFER' && !vm.bankForm.$valid)) {
+        return;
       }
 
+      ga('send', 'event', 'donation', 'donateFromCheckout', $stateParams.projectCode, vm.amount/100);
 
-      $uibModal.open({
-        templateUrl: '/components/checkout/donationConfirmationModal.html',
-        controller: ['$scope', '$state', '$timeout', 'donation', 'projectCode', 'projectTitle', 'HOST', 'AuthService', function ($scope, $state, $timeout, donation, projectCode, projectTitle, HOST, AuthService) {
-          $scope.donation = donation;
-          $scope.loggedUser = AuthService.getLoggedUser();
-          $scope.host = HOST;
-          $scope.projectTitle = projectTitle;
-          $scope.projectShareLink = HOST + 'redirection/project-' + projectCode + '.html';
-          $scope.returnToAppeal = function () {
-            $scope.$dismiss();
-            $state.go('project', {projectCode: projectCode});
-          };
-          $scope.projectCode = projectCode;
-          $timeout(function () {
-            FB.XFBML.parse();
-          });
-        }],
-        resolve: {
-          donation: function () {
-            return vm.amount;
-          },
-          projectCode: function () {
-            return vm.project.code;
-          },
-          projectTitle: function () {
-            return vm.project.title;
-          }
-        }
-      });
-    };
 
-    var successfullyDonated = function (donation) {
-      return donation.status == 'CREATED' || donation.status == 'BIG_TRANSFER_CREATED';
-    };
+      $scope.$dismiss();
 
-    var checkDonationStatus = function (donationId) {
-      $http.get(API + 'checkDonationStatus/' + donationId).then(function (result) {
-        var donation = result.data;
-        console.log(donation);
-        if (successfullyDonated(donation)) {
-          postDonationConfirmation();
-        } else if (result.data.status == 'FAILED') {
-          NotificationService.error("Your bank verification was incorrect – please try again. If you keep getting this error, contact your credit card issuer to resolve the issue.");
-          vm.mode = 'CARD';
+      //TODO: Consider moving that to the checkout service as well
+      if (!AuthService.getLoggedUser()) {
+        AuthService.showLogInModal(CheckoutService.showQuestion);
+      } else {
+        if (vm.mode == 'BANK_TRANSFER') {
+          sendDonation(); // get bank details to show
```

Comment:
Let's use "sendDonationByBankTransfer" here and remove "sendDonation" below

Reply:
That's what I did :)